app/models/concerns/foreman_remote_execution/exportable.rb

```diff
@@ -0,0 +1,54 @@
+# This concern makes it easy to export an ActiveRecord object with specified
+# attributes and associations only.  If a specified assocation also includes
+# this concern, then it will likewise be exported. Custom attributes can be
+# specified as an options hast consisting of a custom export lambda.
+#
+# Example:
+#   attr_exportable :name, :address, :company => ->(user) { user.company.name }
+#
+module ForemanRemoteExecution
+  module Exportable
+    extend ActiveSupport::Concern
```

Comment:
I didn't find a good way to do the kind of selective and recursive exporting of all the associations on JobTemplates, so I made this. The idea here is that you mark say, `template_inputs` as exportable from `JobTemplate` and because `TemplateInput` itself is also exportable, it exports the selected attributes from it too.  Maybe this is crazy :hankey:, it's late.